cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetService.java

```diff
@@ -116,12 +125,15 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     this.discoveryServiceClient = discoveryServiceClient;
     this.opExecutorClient = opExecutorClient;
     this.metricReporters = metricReporters;
+    this.structuredTableAdmin = structuredTableAdmin;
+    this.structuredTableRegistry = structuredTableRegistry;
   }
 
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting DatasetService...");
-
+    structuredTableRegistry.initialize();
+    StoreDefinition.createAllTables(structuredTableAdmin, structuredTableRegistry);
```

Comment:
why should DatasetService create all tables? I thought we added that already to the master and the standalone main, at startup.

Reply:
In distributed mode, will the tables exist if datasetservice is running isolated from master? If so I'll remove from here

Reply:
yes